webapp/src/shared/hooks/UseNavigation.tsx

```diff
@@ -0,0 +1,28 @@
+import { NavigateFunction, useNavigate } from 'react-router-dom';
+
+type Callback = () => void;
+
+type UseNavigationReturn = {
+  navigate: NavigateFunction;
+  goBack: () => Callback;
+  goBackTo: (route: string) => Callback;
+};
+
+export const useNavigation = (): UseNavigationReturn => {
+  const navigate = useNavigate();
+
+  const goBack = () => {
+    return () => {
+      navigate(-1);
+    };
+  };
+
+  const goBackTo = (route: string) => {
+    return () => {
+      navigate(-1);
+      navigate(route);
```

Comment:
Since we are navigating to another route, we have to first pop out of the stack history previous location

Reply:
question: When do we need this behavior to go back and to another page?

Reply:
Because the authUser could change its `username` and, since routes are now based on username, the history stack would point to the old username route

Reply:
(despite the explanation we don't have this anymore, see comment bellow)